### PR TITLE
Reruns of example notebooks

### PR DESCRIPTION
**Pull request by hlapp**
There is one code change here, separated out into 015f7de90d2deea74cb3fe03072b7017d5b456ed, and it only changes which device (GPU or CPU) is used, so should be wholly inconsequential for the results.

The changes in the results are, however, considerable, and presumably reflect the current model having changed from BioCLIP1 to BioCLIP2.

Specifically, the baseline predictive accuracy of BioCLIP2 is considerably better, leaving much less room for improvement for the few-shot classifiers trained on top of BioCLIP embeddings.

We should consider changing the dataset to a more challenging one. For example, we could use a Peromyscus dataset (as in the Imageomics Conference Tool Workshop tutorial), or some Carabids.

### CODE REVIEW

**Comment by hlapp** [comment]
@egrace479 and @thompsonmj there isn't so much to review here in terms of code review (because there aren't many code changes to speak of). Instead, it's more about reviewing and considering the changes in the results, and what actions (regarding example data etc) to take or not as a result.

**Comment by egrace479** [comment]
I'm not sure the original source for the [bird images](https://huggingface.co/datasets/Somnath01/Birds_Species) (that HF dataset doesn't have any context). The 2 of the 4 [iNat images](https://www.inaturalist.org/observations?place_id=any&user_id=john4866&verifiable=any) that are research-grade, shouldn't be in TreeOfLife-200M, since they were ID'd in May 2024 (I believe snapshot was beginning of month).

It might make more sense to use examples from the [testing data](https://huggingface.co/imageomics/bioclip-2#testing-data) for which we could hopefully show meaningful improvement.

**Comment by hlapp** [comment]
I should probably tag @vimar-gu here too for good choices of data where BioCLIP2 has some challenges.

**Comment by vimar-gu** [comment]
What is the size of the data that you are looking for? If it's not a large scale, we can see that BioCLIP 2 cannot tell the zebra species very accurately.

**Comment by hlapp** [comment]
> What is the size of the data that you are looking for?

Not large. Small enough for embedding  the images and training the classifier on the embedding features to each complete in under 10 minutes on a CPU (and thus in ~1-3  minutes on a GPU), and large enough for enough samples per class to meaningfully train and test.

**Comment by hlapp** [comment]
I don't fully remember how large the Peromyscus dataset was that @thompsonmj used for the tutorial, but that seemed to be a good size.

**Comment by thompsonmj** [comment]
I found >10k but filtered down to 1.2k Peromyscus images, ~455M on disk. Can put these somewhere for retrieval into the notebook.

**Comment by thompsonmj** [comment]
If we're looking for challenges, maybe we can do a single HF dataset and retrieve from it in a notebook like:

```
from datasets import load_dataset

ds = load_dataset("Imagoemics/bioclip-trip-ups", "deer-mice")
ds = load_dataset("Imagoemics/bioclip-trip-ups", "zebras")
# etc.
```
as [configurations](https://huggingface.co/docs/datasets/repository_structure#configurations).

**Comment by egrace479** [comment]
> If we're looking for challenges, maybe we can do a single HF dataset and retrieve from it in a notebook like:
> 
> ```
> from datasets import load_dataset
> 
> ds = load_dataset("Imagoemics/bioclip-trip-ups", "deer-mice")
> ds = load_dataset("Imagoemics/bioclip-trip-ups", "zebras")
> # etc.
> ```
> 
> as [configurations](https://huggingface.co/docs/datasets/repository_structure#configurations).

I like this idea. We should also just remember to specify which data are from the training set, e.g., the mice.

**Comment by hlapp** [comment]
Yes indeed I like this idea too, and makes it super straightforward for a tutorial.

**Comment by thompsonmj** [comment]
Any better ideas on what to call the dataset repo?

**Comment by egrace479** [comment]
> Any better ideas on what to call the dataset repo?

1. pybioclip-examples
2. pybioclip-tutorials
3. bioclip-example-tuning
4. bioclip-tuning-tutorials

**Comment by hlapp** [comment]
My vote would be on (3), but in a permutation: `bioclip-tuning-examples`.

**Comment by thompsonmj** [comment]
Such datasets could be useful for tutorials beyond BioCLIP stuff though for other tooling. Maybe something like `fine-grained-challenges`? Something that generalizes to SAEs or Finer-CAM or other tools as well?

**Comment by vimar-gu** [comment]
I agree. All our tools can be applied haha

**Comment by hlapp** [comment]
@egrace479 @thompsonmj I think we need to decide how to move forward here. Specifically, merge these notebook reruns as they are, for the time being resulting in examples that support much less well what we say in the documentation about what they show, or make replacing the dataset part of this change set now.

**Comment by hlapp** [comment]
I'm in favor of the latter (replace dataset now), but I'll need either of you to assist with putting up the dataset in the appropriate place.

**Comment by egrace479** [comment]
> I'm in favor of the latter (replace dataset now), but I'll need either of you to assist with putting up the dataset in the appropriate place.

I agree that we should make replacing the dataset part of this change. 

If we want to pull from the test data, [meta-album](https://meta-album.github.io/) could be a good choice (it wouldn't require us to set something up now); [Med Leaf](https://meta-album.github.io/datasets/MED_LF.html) is a reasonable size.

Regardless, I can make the dataset repo; did we want `fine-grained-challenges`, since it can be used more broadly? Or did we want to stick to the BioCLIP focus with `bioclip-tuning-examples`?

We can add the 1.2k Peromyscus images (clarifying they're part of the training data), @vimar-gu did you have a zebra set?

**Comment by vimar-gu** [comment]
Dan only sent me fewer than 100 images. But if you were talking about the subset from training data, yes, I was using them to train DINO linear probing.

**Comment by thompsonmj** [comment]
Quick note—I'm trying out [Lance](https://huggingface.co/lance-format) ([docs](https://huggingface.co/docs/hub/en/datasets-lance)) for the cryptic species groups challenge set. I'll add a column for `role` too that indicates whether it's part of the model training or if it was held out.

To live in https://huggingface.co/datasets/imageomics/fine-grained-challenges, starting with TOL Peromyscus images.
